arrows/vxl/high_pass_filter.cxx

```diff
@@ -0,0 +1,478 @@
+    {
+      return nullptr;
+    }
+
+    // Only report the summation of directional filterings, contained in the
+    // last plane
+    if( output_net_only && output.nplanes() > 1 )
+    {
+      output = vil_plane( output, output.nplanes() - 1 );
+    }
+
+    return std::make_shared< vxl::image_container >( output );
+  }
+};
+
+// ----------------------------------------------------------------------------
+high_pass_filter
+::high_pass_filter()
+  : d( new priv( this ) )
+{
+  attach_logger( "arrows.vxl.high_pass_filter" );
+}
+
+// ----------------------------------------------------------------------------
+high_pass_filter
+::~high_pass_filter()
+{
+}
+
+// ----------------------------------------------------------------------------
+vital::config_block_sptr
+high_pass_filter
+::get_configuration() const
+{
+  // get base config from base class
+  vital::config_block_sptr config = algorithm::get_configuration();
+
+  config->set_value( "mode", d->mode,
+                     "Operating mode of this filter, possible values: box, bidir" );
+  config->set_value( "kernel_width", d->kernel_width,
+                     "Pixel width of smoothing kernel" );
+  config->set_value( "kernel_height", d->kernel_height,
+                     "Pixel height of smoothing kernel" );
+  config->set_value( "treat_as_interlaced", d->treat_as_interlaced,
+                     "Process alternating rows independently" );
+  config->set_value( "output_net_only", d->output_net_only,
+                     "If set to false, the output image will contain multiple "
+                     "planes, each representing the modal filter applied at "
+                     "different orientations, as opposed to a single plane "
+                     "image representing the sum of filters applied in all "
+                     "directions." );
+
+  return config;
+}
+
+// ----------------------------------------------------------------------------
+void
+high_pass_filter
+::set_configuration( vital::config_block_sptr in_config )
+{
+  // Starting with our generated vital::config_block to ensure that assumed
+  // values are present. An alternative is to check for key presence before
+  // performing a get_value() call.
+  vital::config_block_sptr config = this->get_configuration();
+  config->merge_config( in_config );
+
+  // Settings for filtering
+  d->mode = config->get_value< std::string >( "mode" );
+  d->kernel_width = config->get_value< size_t >( "kernel_width" );
+  d->kernel_height = config->get_value< size_t >( "kernel_height" );
+  d->treat_as_interlaced = config->get_value< bool >( "treat_as_interlaced" );
+  d->output_net_only = config->get_value< bool >( "output_net_only" );
+}
+
+// ----------------------------------------------------------------------------
+bool
+high_pass_filter
+::check_configuration( vital::config_block_sptr config ) const
+{
+  std::string mode_string = config->get_value< std::string >( "mode" );
+
+  if( !( mode_string == "box" ||
```

Comment:
optional impl. No need to validate config item. That is done by the converter/when the config item is converted from string to enum.